src/wp-includes/post.php

```diff
@@ -2289,7 +2296,19 @@ function add_post_type_support( $post_type, $feature, ...$args ) {
 	$features = (array) $feature;
 	foreach ( $features as $feature ) {
 		if ( $args ) {
-			$_wp_post_type_features[ $post_type ][ $feature ] = $args;
+			// Check if feature already exists with args and if both are associative arrays that should be merged.
+			if ( isset( $_wp_post_type_features[ $post_type ][ $feature ][0] )
+				&& is_array( $_wp_post_type_features[ $post_type ][ $feature ][0] )
+				&& isset( $args[0] )
+				&& is_array( $args[0] ) ) {
+				// Merge the associative arrays to preserve existing properties.
+				$_wp_post_type_features[ $post_type ][ $feature ][0] = array_merge(
+					$_wp_post_type_features[ $post_type ][ $feature ][0],
+					$args[0]
+				);
```

Comment:
Nit: This is a bit easier to read and I think is more common formatting for multi-line conditions:

```suggestion
			if (
				isset( $_wp_post_type_features[ $post_type ][ $feature ][0] ) &&
				is_array( $_wp_post_type_features[ $post_type ][ $feature ][0] ) &&
				isset( $args[0] ) &&
				is_array( $args[0] )
			) {
				// Merge the associative arrays to preserve existing properties.
				$_wp_post_type_features[ $post_type ][ $feature ][0] = array_merge(
					$_wp_post_type_features[ $post_type ][ $feature ][0],
					$args[0]
				);
```